docs/docs/javascript/overview/eslint.mdx

```diff
@@ -0,0 +1,554 @@
+
+# Limit warnings to control build failures
+args = ["--max-warnings=0"]
+```
+
+#### Output Control
+```toml title="pants.toml"
+[eslint]
+# Use different output formats
+args = ["--format=compact"]  # Options: stylish, compact, checkstyle, json, junit, tap, unix
+
+# Write reports to files
+args = ["--output-file=reports/eslint-report.json", "--format=json"]
+
+# Quiet mode - only show errors, not warnings
+args = ["--quiet"]
+```
+
+#### File Handling
+```toml title="pants.toml"
+[eslint]
+# Add custom ignore patterns
+args = ["--ignore-pattern=**/dist/**", "--ignore-pattern=**/build/**"]
+
+# Handle unmatched patterns gracefully
+args = ["--no-error-on-unmatched-pattern"]
+
+# Report unused disable directives
+args = ["--report-unused-disable-directives"]
+```
+
+#### Fix Control
+```toml title="pants.toml"
+[eslint]
+# Control which types of fixes to apply (for fmt goal)
+args = ["--fix-type=problem,suggestion"]  # Options: directive, problem, suggestion, layout
+```
+
+#### Configuration Examples
+
+**Performance-optimized for large projects:**
+```toml title="pants.toml"
+[eslint]
+args = [
+    "--cache",
+    "--cache-location=.eslintcache",
+    "--max-warnings=0",
+    "--quiet"
+]
+```
+
+**CI/CD-friendly configuration:**
+```toml title="pants.toml"
+[eslint]
+args = [
+    "--format=json",
+    "--output-file=reports/eslint.json",
+    "--cache",
+    "--max-warnings=0",
+    "--no-error-on-unmatched-pattern"
+]
+json_output = true  # Enables summary parsing in console
+```
+
+**Development configuration with detailed reporting:**
+```toml title="pants.toml"
+[eslint]
+args = [
+    "--cache",
+    "--report-unused-disable-directives",
+    "--format=stylish"
+]
+```
+
+:::tip Cache Location
+ESLint will create a `.eslintcache` file to store cache data. Consider adding this to your `.gitignore`:
+
+```gitignore title=".gitignore"
+.eslintcache
+reports/
+```
+:::
+
+### Common Issues
+
+1. **ESLint not finding config files**: Ensure config files are in the project root or use explicit config path with `config = "path/to/eslint.config.js"`
+2. **TypeScript errors**: Install TypeScript ESLint plugins using `extra_dev_dependencies` and configure parser in your ESLint config
+3. **Performance issues**: Enable ESLint caching with `args = ["--cache"]` for faster subsequent runs
+4. **Cache invalidation**: Delete `.eslintcache` if you encounter stale results after config changes
+5. **Build failures on warnings**: Use `args = ["--max-warnings=0"]` to fail builds on any warnings, or `args = ["--quiet"]` to suppress warnings
+6. **Empty file sets**: Use `args = ["--no-error-on-unmatched-pattern"]` to prevent failures when no files match patterns
+7. **Output format issues**: Use `json_output = true` for parsed summaries, or `args = ["--format=json"]` for raw JSON output
+
+### Debug Mode
+
+Enable debug logging to troubleshoot issues:
+
+```bash
+$ pants --level=debug lint src/js/::
+```
+
+### Custom ESLint Rules
```

Comment:
Again, if this is standard eslint setup - I don't think we need to document it here - just link to their docs.